compiler/il/OMRNode.cpp

```diff
@@ -4597,40 +4597,36 @@ OMR::Node::freeExtensionIfExists()
       }
    }
 
-
-
 TR::DataType
 OMR::Node::getArrayCopyElementType()
    {
-   TR::Compilation * comp = TR::comp();
-   TR_ASSERT(self()->getOpCodeValue() == TR::arraycopy  || self()->getOpCodeValue() == TR::arrayset, "assertion failure");
+   TR_ASSERT(self()->getOpCodeValue() == TR::arraycopy || self()->getOpCodeValue() == TR::arrayset, "getArrayCopyElementType called on an invalid node");
 
-   if (_numChildren==3 || _numChildren==4 || _numChildren==6)
+   if (_numChildren == 3)
```

Comment:
No, we should have exactly two types of arraycopies. One with 3 children and one with 5 children.